src/Pack/Config/TOML.idr

```diff
@@ -21,18 +21,24 @@ export
 FromTOML Codegen where
   fromTOML = tmap Types.fromString
 
+extractString : (errMsg : String) -> (v : Value) -> Either TOMLErr String
+extractString _ (VString s) = Right s
+extractString errMsg _ = Left $ WrongType [] errMsg
+
 export
 FromTOML RlwrapConfig where
   fromTOML _ (VBoolean x)  = Right $ if x then UseRlwrap [] else DoNotUseRlwrap
   fromTOML _ (VString str) = Right $ UseRlwrap [NoEscape str]
   fromTOML _ (VArray xs)   =
-    map (UseRlwrap . fromStrList) $ for xs $
-      \case
-        VString s => Right s
-        _         => Left $ WrongType [] "array of strings"
-
+    map (UseRlwrap . fromStrList) $ traverse (extractString "array of strings") xs
   fromTOML _ _ = Left $ WrongType [] "boolean, string or array of strings"
 
+export
+FromTOML ExtraIdris2ArgsConfig where
+  fromTOML _ (VString str) = Right $ PassExtraArgs [NoEscape str]
```

Comment:
I think, this string should be escaped, i.e. `Escape` rather than `NoEscape`

Reply:
I mean, just imagine what a mess would be it a user accidentally leaves, say, `&` symbol there

Reply:
I modeled this after the instance for `RlwrapConfig`, which also does `NoEscape`. Also, won't escaping mess up spaces?

Reply:
Oh, I see what you mean. Right, the logic with rlwrap config was that if a single string is given, then the string is raw, and spaces in it mean several options at once.

Now I see that ideally we should have been doing it slightly differently, e.g. splitting this string by spaces, and then escaping each separate string (but we shouldn't lose quoted strings with this). But this is irrelevant to this particular PR and should be addressed later.